### DISCUSSION TITLE
How can I have a variable for my site name and build external URLs within documentation?

### DISCUSSION THREAD

**Discussion by nh916** [squidfunk/mkdocs-material#5775] [Q&A]
I want to build URLS with something like `[my link](my_site_variable/about/)` instead of `[my link](https://my_website.com)`

How can I do this within the documentation.

I tried putting in a `site_name` variable at the top of the YAML file that is gloabl.
```yaml
site_url: https://my-website.org
```

and I tried putting it in the `extra` section
```yaml
extra:
  homepage: https://my-website.org
```

and I tried linking to them in my documentation like
```md
[my link]({{ extra.homepage}}/about/)
```

and 
```md
[my link]({{ homepage}}/about/)
```

however, both did not recognize it as a variable and tried taking me to my local site.

Is there a way that I can get this to work for me?

**Comment by squidfunk** [comment]
I think you might be looking for the [`macros`](https://mkdocs-macros-plugin.readthedocs.io/) plugin.

**Reply by mks-d** [reply]
And for those using GitHub Actions for their MkDocs site deployment, this (example [here](https://github.com/ozone-his/ozone-docs/blob/ee3add3adb8852c6bdd32325917fc0da89faf213/.github/workflows/ci.yml#L23-L24)) is needed in their ci.yml file since the plugin must be installed beforehand:
```yml
- run: pip install mkdocs-macros-plugin
```
